app/controllers/return_controller.js

```diff
@@ -6,12 +6,16 @@ const logger = require('winston')
 // Local dependencies
 const Charge = require('../models/charge')
 const responseRouter = require('../utils/response_router')
-const CORRELATION_HEADER = require('../../config/correlation_header').CORRELATION_HEADER
-const withAnalyticsError = require('../utils/analytics').withAnalyticsError
+const { CORRELATION_HEADER } = require('../../config/correlation_header')
+const { withAnalyticsError } = require('../utils/analytics')
+const cookies = require('../utils/cookies')
 
 exports.return = (req, res) => {
   const correlationId = req.headers[CORRELATION_HEADER] || ''
   const charge = Charge(correlationId)
+  // Remove the charge data from the cookie
+  const cookieKey = `ch_${req.chargeId}`
```

Comment:
should this use session#createChargeIdSessionKey to make sure this doesn't get out of sync if how we construct this key changes?

Reply:
👍